migration/manager.py

```diff
@@ -170,20 +170,32 @@ def update_tool_tab(self, tab: ExternalToolTab, is_hidden: bool, position: int |
         logger.debug(result)
         return CourseManager.convert_data_to_tool_tab(result)
 
-    def replace_tool_tab(self, source_tab: ExternalToolTab, target_tab: ExternalToolTab) -> None:
-        target_tab_params: dict[str, Any] = {"hidden": False}
-        if not source_tab.is_hidden:
-            target_tab_params.update({"position": source_tab.position})
-
-        if not source_tab.is_hidden:
-            target_position = source_tab.position
+    def replace_tool_tab(
+        self, source_tab: ExternalToolTab, target_tab: ExternalToolTab
+    ) -> tuple[ExternalToolTab, ExternalToolTab]:
+        logger.debug([source_tab, target_tab])
+
+        # Source tool is hidden in course, don't do anything
+        if source_tab.is_hidden:
+            logger.debug(f'Skipping replacement for {[source_tab, target_tab]}; source tool is hidden.')
+            return (source_tab, target_tab)
         else:
-            target_position = None
-
-        new_target_tab = self.update_tool_tab(tab=target_tab, is_hidden=False, position=target_position)
-        new_source_tab = self.update_tool_tab(tab=source_tab, is_hidden=True)
-        logger.info(f"Successfully replaced tool in course's navigation: {[new_source_tab, new_target_tab]}")
-        return
+            if not target_tab.is_hidden:
+                logger.warning(
+                    f'Both tools ({[source_tab, target_tab]}) are currently available. ' +
+                    'Rolling back will hide the target tool!'
+                )
```

Comment:
WRT rolling back, is that done simply by swapping the IDs of the source/target tools?

Reply:
Yup